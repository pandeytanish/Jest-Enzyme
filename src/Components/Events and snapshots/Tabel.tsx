import {Component} from 'react'

export default class Tabel extends Component {
  render() {
    return (
      <table className='table'>
        <thead>
            <tr>
                <th>Name</th>
                <th>Email</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>Demo</td>
                <td>NA</td>
            </tr>
        </tbody>

      </table>
    )
  }
}
