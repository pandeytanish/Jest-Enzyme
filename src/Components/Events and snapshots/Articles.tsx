import React, { Component } from 'react'
import Tabel from './Tabel'
interface State {
    count : number
}

export default class Articles extends Component<{},State> {
    constructor(props : {}) {
        super(props)
        this.state = {
            count : 0
        }
    }
  render() {
    return (
      <div>
        <div className='click-1'></div>
        <div className={`click-${this.state.count}`}>
            {this.state.count} clicks
        </div>
        <button onClick={() => {this.setState({count : this.state.count + 1})}}>Increment</button>
        <h1>Article Details</h1>
        <Tabel/>
      </div>
    )
  }
}
