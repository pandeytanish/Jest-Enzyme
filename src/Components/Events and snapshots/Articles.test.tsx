import Articles from "./Articles";
import { shallow, mount} from "enzyme";
import Tabel from "./Tabel";

describe.skip('Article Page', () => {
    test('click count', () => {
        let wrapper = shallow(<Articles/>)
        expect(wrapper.find('.click-0').length).toEqual(1)
        wrapper.find('button').simulate('click');
        expect(wrapper.find('.click-1').length).toEqual(2)
    })

    test('click 3 count', () => {
        let wrapper = shallow(<Articles/>)
        expect(wrapper.find('.click-0').length).toEqual(1)
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        expect(wrapper.find('.click-2').length).toEqual(1)
    })

    test('dive', () => {
        let wrapper = shallow(<Articles/>)
        expect(wrapper.find(Tabel).dive().find('.table').length).toBe(1)
        
    })

    test('snapshot', () => {
        let wrapper = shallow(<Tabel/>)
        expect(wrapper).toMatchSnapshot();
    })
    test('snapshot 2', () => {
        let wrapper = shallow(<Tabel/>)
        expect(wrapper).toMatchSnapshot();
    })
    test('snapshot 3', () => {
        let wrapper = shallow(<Tabel/>)
        expect(wrapper).toMatchSnapshot();
    })
})

describe('mount', () => {
    test('mount test', () => {
        let wrapper = mount(<Articles/>)
        console.log(wrapper.debug())
    })
})