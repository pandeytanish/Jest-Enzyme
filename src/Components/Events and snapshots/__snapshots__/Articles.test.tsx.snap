// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Page snapshot 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Article Page snapshot 2 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Article Page snapshot 3 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;
